modular_hearthstone/code/modules/spells/roguetown/wizard.dm

```diff
@@ -1539,7 +1539,124 @@
 		REMOVE_TRAIT(H, TRAIT_MIRROR_MAGIC, TRAIT_GENERIC)
 		to_chat(H, span_warning("Your connection to mirrors fades away."))
 
+/obj/effect/proc_holder/spell/invoked/shadowstep
+	name = "Shadowstep"
+	desc = "Project your shadow to swap places with it, teleporting several feet away."
+	cost = 1
+	xp_gain = TRUE
+	releasedrain = 30
+	warnie = "spellwarning"
+	movement_interrupt = TRUE
+	associated_skill = /datum/skill/magic/arcane
+	overlay_state = "shadowstep"
+	chargedrain = 1
+	chargetime = 0 SECONDS
+	charge_max = 30 SECONDS
+	var/area_of_effect = 1
+	var/max_range = 7
+	var/turf/destination_turf
+	var/turf/user_turf
+	var/mutable_appearance/tile_effect
+	var/leash_range = 7
+	var/mutable_appearance/target_effect
+	var/datum/looping_sound/invokeshadow/shadowloop
+
+//Resets the tile and turf effects.
+/obj/effect/proc_holder/spell/invoked/shadowstep/proc/reset(silent = FALSE)
+	if(tile_effect && destination_turf)
+		destination_turf.cut_overlay(tile_effect)
+		qdel(tile_effect)
+		destination_turf = null
+	if(user_turf && target_effect)
+		user_turf.cut_overlay(target_effect)
+		qdel(target_effect)
+		user_turf = null
+	update_icon()
+
+/obj/effect/proc_holder/spell/invoked/shadowstep/proc/check_path(turf/Tu, turf/Tt)
```

Comment:
This proc doesn't account for a variety of other obstacles in the way. It would allow someone to break into the stewardry, or merchant's place behind the counter / bars, or past a closed Keep gate.
I also believe one could alt-click any door (like the Lord's Chamber) and teleport to the turf under the door, and thus gain access that way.

Reply:
I'm not saying if they should or shouldn't, though, I don't really have strong opinions one way or the other.

Reply:
The first I don't mind so much, IC and probably ooc consequences if abused in such a way. But teleporting ontop of a door is very gamey.. I think I can make it check for a door object.

Reply:
> The first I don't mind so much, IC and probably ooc consequences if abused in such a way. But teleporting ontop of a door is very gamey.. I think I can make it check for a door object.

Same thing for windows. In both cases, you might want to account for them being open. I dunno if that'd be their density var or something bespoke like with windows `climbable` var.
(I kinda deliberately picked my grappler code to only go upwards to avoid 90% of this kind of shenanigans)

Reply:
patched woo